File: GO/Mamiferos.go
package main

// Interface para los métodos abstractos
type Mamiferos interface {
	Amamantar() string
}

// Struct “base” que simula la clase padre
type BaseMamiferos struct {
	Nombre string
	Edad int
}

// Constructor para BaseMamiferos
func NewBaseMamiferos(nombre string, edad int) *BaseMamiferos {
	return &BaseMamiferos{
		Nombre: nombre, 
		Edad: edad,
	}
}

// Simuacion de metodos getters
func (b *BaseMamiferos) GetNombre() string {
    return b.Nombre
}

func (b *BaseMamiferos) GetEdad() int {
    return b.Edad
}

File: C#/C#/clases_abstractas/Mamiferos.cs
using System;

namespace C_.clases_abstractas
{
    internal abstract class Mamiferos
    {
        public String nombre;
        public byte edad;

        // Constructor con parametros
        public Mamiferos(String nombre, byte edad)
        {
            this.nombre = nombre;
            this.edad = edad;
        }

        // Metodos
        public abstract void amamantar();
    }
}
